pkg/scheduler/core/generic_scheduler.go

```diff
@@ -146,6 +152,24 @@ func (g *genericScheduler) snapshot() error {
 // If it succeeds, it will return the name of the node.
 // If it fails, it will return a FitError error with reasons.
 func (g *genericScheduler) Schedule(ctx context.Context, prof *profile.Profile, state *framework.CycleState, pod *v1.Pod) (result ScheduleResult, err error) {
+	annotation, ok := pod.Annotations["trace.kubernetes.io/context"]
+	if ok {
+		propagator := trace.DefaultHTTPPropagator()
+		reqHeader := &http.Header{}
+		decodedContextBytes, err := base64.StdEncoding.DecodeString(annotation)
+		if err != nil {
+			return result, err
+		}
+		err = json.Unmarshal(decodedContextBytes, reqHeader)
+		if err != nil {
+			return result, err
+		}
+		ctx = propagator.Extract(ctx, reqHeader)
+	}
+	ctx, span := global.TraceProvider().Tracer("new-scheduler").Start(ctx, fmt.Sprintf("genericScheduler: Schedule '%s'", pod.Name))
+	span.SetAttributes(core.KeyValue{Key: "PodName", Value: core.String(pod.Name)})
```

Comment:
it's adviseable to check for `IsRecording` before setting attributes: https://github.com/open-telemetry/opentelemetry-go/blob/1301b6f3e4b5a8c903e86dbc030e0527d01f807f/api/trace/api.go#L100-L101